### PR TITLE
Add ImportedObject model to importer resources.

### PR DESCRIPTION
**Pull request by szok**

### CODE REVIEW

File: src/ralph/data_importer/widgets.py [hunk 1]
```diff
@@ -6,10 +6,33 @@
...
+            content_type = ContentType.objects.get_for_model(self.model)
+            try:
+                imported_obj = ImportedObjects.objects.get(
+                    content_type=content_type,
+                    old_object_pk=int(value)
+                )
+                try:
+                    return self.model.objects.get(pk=imported_obj.object_pk)
+                except self.model.DoesNotExist:
+                    return None
```
Comment (mkurek) [inline src/ralph/data_importer/widgets.py]: add some logging here

File: src/ralph/data_importer/resources.py [hunk 2]
```diff
@@ -240,7 +248,7 @@ class DiskShareMountResource(resources.ModelResource):
     share = fields.Field(
         column_name='share',
         attribute='share',
-        widget=widgets.ForeignKeyWidget(DiskShare, 'wwn'),
+        widget=widgets.ForeignKeyWidget(ImportedForeignKeyWidget),
```
Comment (mkurek) [inline src/ralph/data_importer/resources.py]: it's by purpose or a mistake? Shouldn't it be `widget=ImportedForeignKeyWidget(DiskShare),`?

File: src/ralph/data_importer/models.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+@python_2_unicode_compatible
+class ImportedObjects(TimeStampMixin, models.Model):
+
+    """Django models for imported objects."""
+
+    content_type = models.ForeignKey(ContentType)
+    object_pk = models.IntegerField(db_index=True)
+    old_object_pk = models.IntegerField(db_index=True)
```
Comment (ar4s) [inline src/ralph/data_importer/models.py]: `old_object_pk` should be `CharField`

**Comment by ghost** [comment]
I think not every ForeignKey is replaced with ImportedForeignKeyWidget.

File: src/ralph/data_importer/mixins.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+from ralph.data_importer.models import ImportedObjects
+
+
+class ImportForeignKeyMixin(object):
+
+    """ImportForeignKeyMixin class for django import-export resources."""
+
+    def before_save_instance(self, instance, dry_run):
+        self.old_object_pk = instance.pk
+        instance.pk = None
```
Comment (mkurek) [inline src/ralph/data_importer/mixins.py]: what about importing with fixed id/pk? Didn't we discuss it that pk shouldn't be cleaned and old ralph should export id in other column?

File: src/ralph/data_importer/widgets.py [hunk 5]
```diff
@@ -5,11 +5,41 @@
...
+            imported_obj = ImportedObjects.objects.filter(
+                content_type=content_type,
+                old_object_pk=value
+            ).first()
+            if imported_obj:
+                result = self.model.objects.filter(
+                    pk=imported_obj.object_pk
+                ).first()
+            else:
+                logger.debug("Not found record pk:{pk} for {model}".format(
```
Comment (mkurek) [inline src/ralph/data_importer/widgets.py]: warning instead of debug

Comment (mkurek) [inline src/ralph/data_importer/widgets.py]: `Record with pk {pk} not found for model {model}`

File: src/ralph/data_importer/widgets.py [hunk 6]
```diff
@@ -5,11 +5,41 @@
...
+        if value:
+            content_type = ContentType.objects.get_for_model(self.model)
+            imported_obj = ImportedObjects.objects.filter(
+                content_type=content_type,
+                old_object_pk=value
+            ).first()
+            if imported_obj:
+                result = self.model.objects.filter(
+                    pk=imported_obj.object_pk
+                ).first()
```
Comment (xliiv) [inline src/ralph/data_importer/widgets.py]: Are you sure this code is correct?

**Comment by mkurek** [comment]
:ok:

**Comment by ghost** [comment]
:ok:

**Pull request merged**